src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -47,19 +53,33 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
             var initialNode = root.FindNode(diagnostic.Location.SourceSpan);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
 
             for (var node = initialNode; node != null; node = node.Parent)
             {
-                if (node is TObjectCreationExpressionSyntax objectCreation)
-                {
-                    var argumentOpt = TryGetRelevantArgument(initialNode, node, diagnostic);
-                    await HandleObjectCreationExpressionAsync(context, objectCreation, argumentOpt).ConfigureAwait(false);
-                    return;
-                }
-                else if (node is TInvocationExpressionSyntax invocationExpression)
+                var fixData =
+                    TryGetInvocationExpressionFixInfo(semanticModel, syntaxFacts, node, cancellationToken) ??
+                    TryGetObjectCreationFixInfo(semanticModel, syntaxFacts, node, cancellationToken) ??
+                    TryGetLanguageSpecificFixInfo(semanticModel, node, cancellationToken);
+
+                if (fixData != null)
                 {
+                    var candidates = fixData.MethodCandidates;
+                    if (fixData.IsConstructorInitializer)
+                    {
+                        // The invocation is a :this() or :base() call. In  the 'this' case we need to exclude the 
```

Comment:
okay. got confused a bit. so what it is saying is, I am going to remove the constructor where this fix is invoked from since otherwise, it can add a parameter to itself.

Reply:
C(): this(1)

would be fixed to 

C(int v):this(1)

This isn't valid C# and gives a compiler error.

Reply:
yep.

I first thought from comment, It is doing this

```csharp
C(int 123invalidName) {}

C() : this(1, 1)
```

since candidate C(int 123invalidName) has error in it, this(1, 1) will not consider adding int v to the C(int 123invalidName)